src/selfmain.c

```diff
@@ -117,9 +117,14 @@ void setBootProt(int v) {
     // 'endWriteIndex' is initialized with bytes that should be written.
     // N.B. every itearation write a quadword, hence may be written more bytes than requested
     const size_t endWriteIndex = format ? sizeof(fuses) : repair_fuses ? sizeof(newfuses) : 4;
-    for (int i = 0; i < endWriteIndex; i += 16) {
-        uint32_t *const qwBlockAddr = (uint32_t *const)(NVM_FUSE_ADDR + i);
-        memcpy(qwBlockAddr, &fuses[i], 16);
+#if defined(SAMD21)
+#define UPDATE_COUNT 2
+#elif defined(SAMD51)
+#define UPDATE_COUNT 16
+#endif
+    for (int i = 0; i < endWriteIndex; i += UPDATE_COUNT) {
+        uint32_t *const qwBlockAddr = ((uint32_t *const)NVM_FUSE_ADDR) + i;
+        memcpy(qwBlockAddr, &fuses[i], UPDATE_COUNT);
         #if defined(SAMD21)
             exec_cmdaddr(NVMCTRL_CTRLA_CMD_WAP, qwBlockAddr);
         #elif defined(SAMD51)
```

Comment:
This looks good, but the `qwBlockAddr` variable name is misleading for the SAMD21 case. Could you rewrite this as a single `#if`-`#elif`, and then just copy the loop into each arm of the `#if`, with the proper `i += ...` and the proper `exec_cmdaddr()`? I think that will be easier to read. Thanks.

Reply:
How about this -- I split the SAMD21 code into a non-looping bit as there are always 8 bytes to write in that case.

Reply:
Looks good, thank you!